docs/concepts.rst

```diff
@@ -0,0 +1,448 @@
+* Completely abstracted organization and management of data files on disk
+* Ability to only retrieve a small portion of the data (as needed) while still
+  maintaining complete historical record
+* Ability to push and pull changes directly to collaborators or a central server
+  (ie a truly distributed version control system)
+
+The ability of version control systems to perform these tasks for codebases is
+largely taken for granted by almost every developer today; However, we are
+in-fact standing on the shoulders of giants, with decades of engineering which
+has resulted in these phenomenally useful tools. Now that a new era of
+"Data-Defined software" is taking hold, we find there is a strong need for
+analogous version control systems which are designed to handle numerical data at
+large scale... Welcome to Hangar!
+
+***********
+Inspiration
+***********
+
+The design of hangar was heavily influenced by the `Git <https://git-scm.org>`_
+source-code version control system. As a Hangar user, many of the fundamental
+building blocks and commands can be thought of as interchangeable:
+
+* checkout
+* commit
+* branch
+* merge
+* diff
+* push
+* pull/fetch
+* log
+
+Emulating the high level the git syntax has allowed us to create a user
+experience which should be familiar in many ways to Hangar users; a goal of the
+project is to enable many of the same VCS workflows developers use for code
+while working with their data!
+
+There are, however, many fundamental differences in how humans/programs
+interpret and use text in source files vs. numerical data which raise many
+questions Hangar needs to uniquely solve:
+
+* How do we connect some piece of "Data" with a meaning in the real world?
+* How do we diff and merge large collections of data samples?
+* How can we resolve conflicts?
+* How do we make data access (reading and writing) convenient for both
+  user-driven exploratory analyses and high performance production systems
+  operating without supervision?
+* How can we enable people to work on huge datasets in a local (laptop grade)
+  development environment?
+
+We will show how hangar solves these questions in a high-level guide below.
+For a deep dive into the Hangar internals, we invite you to check out the
+:ref:`ref-hangar-under-the-hood` page.
+
+****************************
+How Hangar Thinks About Data
+****************************
+
+From this point forward, **when we talk about "Data" we are actually talking
+about n-dimensional arrays of numeric information. To Hangar, "Data" is just a
+collection of numbers being passed into and out of it.** Data does not have a
+file type, it does not have a file-extension, it does not mean anything to
+Hangar itself - it is just numbers. This theory of "Data" is nearly as simple as
+it gets, and this simplicity is what enables us to be unconstrained as we build
+abstractions and utilities to operate on it.
+
+Abstraction 1: What is a Repository?
```

Comment:
Move this later, since you use `Dataset` here, but we don't know what it is yet.  I'd start with `Sample` then `Dataset` then `Repository` since that builds from small to large.